### PR TITLE
Add configQuery to every component for specific config querying #326

### PR DESCRIPTION
**Pull request by pmi**

### CODE REVIEW

File: src/_enonicAdapter/guillotine/fetchContent.ts [hunk 1]
```diff
@@ -529,11 +530,29 @@ function collectComponentDescriptors(components: PageComponent[],
 
 function processComponentConfig(myAppName: string, myAppNameDashed: string, cmp: PageComponent) {
     const cmpData = cmp[cmp.type];
-    if (cmpData && 'descriptor' in cmpData && cmpData.descriptor && 'configAsJson' in cmpData && cmpData.configAsJson) {
+    if (cmpData && 'descriptor' in cmpData && cmpData.descriptor) {
```
Comment (poi33) [inline src/_enonicAdapter/guillotine/fetchContent.ts:533]: `cmpData?.descriptor` ? I don't know the difference of `in` and conditional chaining. It looks the same to me 🤷 
These two look the same to me: `'descriptor' in cmpData && cmpData.descriptor`

File: src/_enonicAdapter/guillotine/fetchContent.ts [hunk 2]
```diff
@@ -529,11 +530,29 @@ function collectComponentDescriptors(components: PageComponent[],
...
         const [appName, cmpName] = cmpData.descriptor.split(':');
-        if (appName === myAppName && cmpData.configAsJson[myAppNameDashed][cmpName]) {
-            cmpData.config = cmpData.configAsJson[myAppNameDashed][cmpName];
-            delete cmpData.configAsJson;
+        if (appName !== myAppName) {
+            return;
+        }
+        let config;
+        const configArray: Object[] = [];
+        if ('configAsJson' in cmpData && cmpData.configAsJson && cmpData.configAsJson[myAppNameDashed]) {
```
Comment (poi33) [inline src/_enonicAdapter/guillotine/fetchContent.ts:540]: `cmpData?.configAsJson && myAppNameDashed in cmpData.configAsJson` ?

**Review by poi33 — APPROVED** [review]
Some minor things that can be improved. Optional

File: src/_enonicAdapter/guillotine/fetchContent.ts (hunk 1 above)
Comment (pmi) [reply src/_enonicAdapter/guillotine/fetchContent.ts:533]: Right, the issue here is cmpData can be of different type including TextData, which doesn't have descriptor field.
Typescript doesn't like that and requires a check for field presence `'descriptor' in cmpData` before we can check it for undefined with `cmpData.descriptor`

File: src/_enonicAdapter/guillotine/fetchContent.ts (hunk 2 above)
Comment (pmi) [reply src/_enonicAdapter/guillotine/fetchContent.ts:540]: Same as above I assume, configAsJson is not present for TextData

File: src/_enonicAdapter/guillotine/fetchContent.ts (hunk 1 above)
Comment (poi33) [reply src/_enonicAdapter/guillotine/fetchContent.ts:533]: Okay i see why this is needed. Keep as is

**Pull request merged**